lang/R/inst/extdata/airr-schema.yaml

```diff
@@ -478,12 +480,74 @@ TimePoint:
                     top_node:
                         id: UO:0000003
                         label: time unit
-
 #
 # General objects
-# TODO: link to global schema with JSON-LD?
 #
 
+# A time range or interval
+TimeInterval:
+    description: Time range or interval for a measurement, observation or action.
+    type: object
+    properties:
+        min:
+            type: number
+            description: Lower/minimum value of the time interval
+            example: 5.0
+            x-airr:
+                adc-query-support: true
+                nullable: true
+        max:
+            type: number
+            description: Upper/maximum value of the time interval
+            example: 10.0
+            x-airr:
+                adc-query-support: true
+                nullable: true
+        unit:
+            $ref: '#/Ontology'
+            description: Unit of the time interval
+            example:
+                id: UO:0000033
+                label: day
+            x-airr:
+                adc-query-support: true
+                nullable: true
```

Comment:
Why is `unit` `nullable: true`? I understand that we need `min` and `max` nullable, but this should always contain a unit of time.

Reply:
They have always been that way since they were added (e.g. collection_time_point_relative_unit was always nullable... 8-)

I suppose in a TimePoint or TimeInterval it makes sense to change this behaviour. Having the unit nullable when it was a field by itself makes some sense, much less sense in the new constructs I agree.

Reply:
good catch, yes the whole object can be null, but if you specify the object then should always have the values and the units. I pushed a commit to change that.